Sources/SharedModels/Course/CoursesForDashboardDTO.swift

```diff
@@ -11,5 +11,5 @@ public struct CoursesForDashboardDTO: Codable {
     public var courses: [CourseForDashboardDTO]?
     public var activeExams: [Exam]?
 
-    public init() {}
```

Comment:
This would be a major change and would break some `#Previews`, I think. Can we keep it?

Reply:
From what I could tell, everything still works. For structs, Swift generates a memberwise initializer by default, in which optionals have a default value of nil (so in this case `init(courses: [CourseForDashboardDTO]? = nil, activeExams: [Exam]? = nil)`, which can then be called as `init()` without issues). I'll double check to make sure, but I'm pretty certain this is what the default init looks like since I also got a warning about the empty init already existing.

Reply:
Looks like both of us were right. The default init does look the way I described, however, it is not public and therefore does change the API. Would you prefer to have a public default init, to have two inits (private default and public without arguments), or the current solution with theoretically breaking API changes? The only usages in Artemis-iOS of this empty init have already been replaced by .mock anyways.

Reply:
Yes, it's not public.

Mocks now exist, but I liked to create my own value during development without checking out core-modules locally

Reply:
Does anything speak against making the default constructor public in that case? Alternatively, using `.mock` as the `init` in such a case would probably have a similar effect if you're changing the member variables after that call anyways. Would prefer to either have a proper init or none over an empty one to be honest. Since we already have breaking changes with UserSessionFactory, it doesn't really matter which approach we go with